In [1]:
import pandas as pd

df = pd.read_csv('charity_data.csv')
pd.set_option('display.max_columns', 23)


*2 ЗАДАЧА*

In [11]:
#перекодировка size : {'small' : 1, 'mid' : 2, 'big' : 3}
df['size'] = df['size'].replace({'small' : 1, 'mid' : 2, 'big' : 3})
df

ascore                   category  \
0      100.0                Environment   
1       89.0  Arts, Culture, Humanities   
2       70.0                   Religion   
3       93.0  Arts, Culture, Humanities   
4      100.0             Human Services   
...      ...                        ...   
8403    92.0  Arts, Culture, Humanities   
8404    59.0                   Religion   
8405   100.0             Human Services   
8406   100.0             Human Services   
8407   100.0  Arts, Culture, Humanities   

                                            description         ein  \
0     Working with Oregonians to enhance our quality...  93-0642086   
1     Serving the metropolitan Baltimore area and th...  31-1770828   
2     VSS Catholic Communications is dedicated to an...  91-1857425   
3     The mission of the Utah Symphony & Opera is to...  51-0145980   
4     Funded solely by the footwear industry, Two Te...  22-2579809   
...                                                 ...         ...   
8403  Founded in 1900, Yaddo is an artists' communit...  14-1343055   
8404  Yad L'Achim was established in 1950. It is a n...  23-7126417   
8405  Yad Ezra opened its doors in 1990 with the pur...  38-2904733   
8406  Rawhide, where commitment and compassion meet....  39-1052471   
8407  The Raymond F. Kravis Center for the Performin...  59-2245054   

         tot_exp  admin_exp_p  fund_eff  fund_exp_p  program_exp_p  fscore  \
0       917313.0        0.101      0.18       0.184          0.714   88.61   
1      5053059.0        0.231      0.05       0.032          0.735   82.85   
2      1891500.0        0.114      0.17       0.197          0.687   86.74   
3     22506582.0        0.092      0.08       0.054          0.852   91.03   
4      5594243.0        0.126      0.12       0.122          0.750   86.23   
...          ...          ...       ...         ...            ...     ...   
8403   3283832.0        0.231      0.22       0.196          0.572   71.94   
8404   1387396.0        0.147      0.27       0.285          0.567   60.71   
8405   2419474.0        0.061      0.04       0.052          0.885   83.13   
8406  14606304.0        0.128      0.12       0.073          0.797   95.64   
8407  31969350.0        0.148      0.19       0.046          0.805   89.10   

                                      leader  leader_comp  leader_comp_p  \
0                              Lynn Peterson          0.0         0.0000   
1                         Anthony S. Brandon     163341.0         0.0323   
2                                Jim Carroll     114791.0         0.0606   
3                               Paul Meecham     124805.0         0.0055   
4                                Neal Newman     219002.0         0.0391   
...                                      ...          ...            ...   
8403                       Elaina Richardson     289028.0         0.0880   
8404  This organization did not report a CEO          NaN            NaN   
8405                               Lea Luger     130000.0         0.0537   
8406                            John Solberg     129314.0         0.0088   
8407                      Judith A. Mitchell     480410.0         0.0150   

                                                  motto  \
0      Great communities. Working lands. Iconic Places.   
1                      88.1 FM -. Your NPR News Station   
2                         Spirit Catholic Radio Network   
3              Engaging, educating, and enriching lives   
4                         Shoepeople Helping Shoepeople   
...                                                 ...   
8403                     Nurturing the creative process   
8404  Promoting the adherence to Orthodox Jewish obs...   
8405                          Feeding the Jewish hungry   
8406  Dedicated to helping at-risk youth and their f...   
8407  Enhancing the quality of life in Palm Beach Co...   

                                                   name     tot_rev  score  \

In [4]:
num_c = df.select_dtypes(exclude=['object'])
#корреляция
correlation_matrix = num_c.corr()
#таблицка с цветами корреляций
correlation_matrix.style.background_gradient(cmap='coolwarm')

Между tot_exp и tot_rev прослеживается высокий коэф. кореляции. При росте первого показателя второй также растет. Тут ничего сложного: растут доходы - растут расходы

Program_exp_p и admin_exp_p имеют высокую отрицательную корреляцию(-0.726408). Понятно что при большом административном расходе программные расходы будут падать. 

Очень странная корреляция size с leader_comp(0.487926) и с leader_comp_p(-0.576583). Как это объяснить - загадка

*1 ЗАДАЧА*

In [9]:
import scipy.stats

In [7]:
#Коэффициент корреляции между fund_eff и fscore, p-value:
scipy.stats.pearsonr(df['fund_eff'], df['fscore'])


PearsonRResult(statistic=-0.5485086095516047, pvalue=0.0)


In [8]:
#Коэффициент корреляции между fund_eff и score, p-value:
scipy.stats.pearsonr(df['fund_eff'], df['score'])

PearsonRResult(statistic=-0.40572683324824077, pvalue=0.0)

Между tot_exp и tot_rev прослеживается высокий коэф. кореляции. При росте первого показателя второй также растет. Тут ничего сложного: растут доходы - растут расходы

Program_exp_p и admin_exp_p имеют высокую отрицательную корреляцию(-0.726408). Понятно что при большом административном расходе программные расходы будут падать. 

Очень странная корреляция size с leader_comp(0.487926) и с leader_comp_p(-0.576583). Как это объяснить - загадка

*3 Задание*

In [26]:
category_mask = df['category'].replace({'Environment':0, 'Arts, Culture, Humanities':1, 'Religion':2,
       'Human Services':3, 'Education':4, 'Animals':5, 'International':6,
       'Health':7, 'Community Development':7, 'Human and Civil Rights':8,
       'Research and Public Policy':9})

category_mask = category_mask.to_frame()
category_mask

category
0            0
1            1
2            2
3            1
4            3
...        ...
8403         1
8404         2
8405         3
8406         3
8407         1

[8408 rows x 1 columns]

In [17]:
# category_mask
score = df['score']
score_and_category = pd.concat([category_mask, score], axis=1)


In [19]:
correlation_score_and_category = score_and_category.corr()
#таблицка с цветами корреляций
correlation_score_and_category.style.background_gradient(cmap='coolwarm')

In [28]:
corr = df['score'].corr(category_mask['category'])
corr

0.05779448347431325

- зависят ли рейтинг (score) и категория благотворительности?

Статистическая гипотеза не подтвердилась. Корреляция оказалась близка к нулю (0.05779448347431325)